libportal/background.c

```diff
@@ -20,9 +20,115 @@
 
 #include "config.h"
 
+#include "session-private.h"
 #include "background.h"
 #include "portal-private.h"
 
+typedef struct {
+  XdpPortal *portal;
+  GTask *task;
+  char *status_message;
+} SetStatusCall;
+
+static void
+set_status_call_free (SetStatusCall *call)
+{
+  g_clear_pointer (&call->status_message, g_free);
+  g_clear_object (&call->portal);
+  g_clear_object (&call->task);
```

Comment:
Should `g_free (call);` too shouldn't it?

Reply:
Correct, my bad